packages/region-config-resolver/src/regionConfig/resolveRegionConfig.ts

```diff
@@ -0,0 +1,68 @@
+import { Provider } from "@smithy/types";
+
+import { getRealRegion } from "./getRealRegion";
+import { isFipsRegion } from "./isFipsRegion";
+
+/**
+ * @public
+ */
+export interface RegionInputConfig {
+  /**
+   * The AWS region to which this client will send requests
+   */
+  region?: string | Provider<string>;
+
+  /**
+   * Enables FIPS compatible endpoints.
+   */
+  useFipsEndpoint?: boolean | Provider<boolean>;
+}
+
+/**
+ * @internal
+ */
+interface PreviouslyResolved {}
+
+/**
+ * @internal
+ */
+export interface RegionResolvedConfig {
+  /**
+   * Resolved value for input config {@link RegionInputConfig.region}
+   */
+  region: Provider<string>;
+
+  /**
+   * Resolved value for input {@link RegionInputConfig.useFipsEndpoint}
+   */
+  useFipsEndpoint: Provider<boolean>;
+}
```

Comment:
the input and resolved config are duplicates to types in smithy/config-resolver, what's the plan for these, co-existence or deprecation of one?

Can you just import the types from there?

Reply:
No because the concept of region is AWS specific and should belong to aws-js-sdk-v3